discovery_engine_core/bindings/src/types/uuid.rs

```diff
@@ -0,0 +1,135 @@
+// Copyright 2022 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+//! FFI functions for handling `Uuid`
+
+use std::ptr;
+
+use uuid::Uuid;
+
+/// Creates a new UUID based on this byes (~ `[u8; 16]`).
+///
+/// The bytes are passed in as separate parameters as dart
+/// can't handle C values on the stack well.
```

Comment:
Don't we need to add constraints on the order and endianes?

Reply:
no it's a sequence of bytes, endianes only matters for integers larger then `u8`

It's the byte sequence representation of an Uuid. And how that maps to a non
byte representation is specified by the standards around Uuids.